### DISCUSSION TITLE
How to model uncertainty about events?

### DISCUSSION THREAD

**Discussion by ManuelSimoes** [nemo-ufes/gufo#28] [Q&A] [answered]
In the domain of law enforcement there is always quite a bit of uncertainty about what happened:
- E.g. two witnesses giving different accounts of the same event. 
- E.g. a cop suspecting that a crime happened. 

If we model these things as a gufo:ExtrinsicMode (suspicion, witness claim, etc) than we lose the structure of what it is that's being claimed. 
If we model these things as true than we're technically lying.

Something like a gufo:ExtrinsicMode (suspicion, witness claim, etc) that refers to a named graph that contains what it is that's being claimed would seem like a solution, but doesn't seem possible with gUFO.

How would you model this type of uncertainty?

**Comment by pedropaulofb** [comment]
Hi @ManuelSimoes, 
This might not be the best forum for questions about modeling strategies. I’m part of a WhatsApp group dedicated to UFO/OntoUML/gUFO, which could be more suitable for this discussion. If you're interested, please send me an email with your phone number and I’ll add you to the group.

**Reply by ManuelSimoes** [reply]
Hey Pedro, I sent you my phone number, and that of a colleague, on July 8 and 11 . Would be great if you could add us.

**Reply by pedropaulofb** [reply]
Hi @ManuelSimoes ! I am so sorry, I did not receive it.

Can you please send me the numbers again to one of these emails: p.p.favatobarcelos@utwente.nl or pedro.favatobarcelos@ugent.be?

**Reply by jpalmeida** [reply]
@pedropaulofb let's encourage public discussions, and engage here instead of in private.

**Reply by pedropaulofb** [reply]
@jpalmeida and @ManuelSimoes ,


Agreed. I’ll inform the members of the group that this forum is the best place for discussions.

**Comment by djradon** [comment]
FWIW, GitHub Discussions Q&A seems like a great forum to me. It's public, structured, and search-engine indexed, so people searching for answers can find relevant discussions. Plus, it's got labels and notification subscriptions.

**Reply by jpalmeida** [reply]
I too prefer for these questions to be addressed here. I think Pedro Paulo's comment is now stale, as it was a reaction to the question that was raised as an issue. @pedropaulofb is that the case?

**Comment by jpalmeida** [comment, marked as answer]
Modes can be intentional in the sense that they may have a propositional content. So, you could have gufo:IntrinsicMode (for example, a Belief, or a Suspicion) whose propositional content is an rdf:Statement [1]. The same with respect to a gufo:ExtrinsicMode (for example, a WitnessClaim in a court). Similarly, if you assume RDF-star [2] instead of an rdf:Statement you could have a quoted triple.

[1] https://www.w3.org/TR/rdf11-schema/#ch_reificationvocab
[2] https://w3c.github.io/rdf-star/cg-spec/editors_draft.html

**Comment by ManuelSimoes** [comment]
Thanks @jpalmeida for elucidating. I have some follow up questions though:

1. The difference between IntrinsicMode and ExtrinsicMode in this context is not quite clear to me, e.g. why is a belief that the Eiffel tower is located in Paris an IntrinsicMode and John’s admiration for Mary an ExtrinsicMode? You could argue that both externally depend on something outside the bearer of the aspect (e.g. the Eiffel Tower, Paris, Mary). You could also argue that both only inhere in the bearer (e.g. the Eiffel tower might not be in Paris, Mary might not exist). How do you decide which mode to use?
2. In the same vein: you state that a suspicion would be an IntrinsicMode and a witness statement in court an ExtrinsicMode. I suppose the latter seems obvious (the witness statement depends on there being a court), although you could also argue it depends more so on there being a court case, which would be an Event and to which externallyDependsOn doesn't apply . The former also doesn't seem obvious: a suspicion that a crime happened (i.e. an Event) could be construed as an IntrinsicMode (as it can’t be externally dependent on an Event), but a suspicion that X commited the crime could be construed as an ExtrinsicMode (externally dependent on X). Intuitively it seems odd that there'd be two different kinds of suspicion, each specializing from a different gufo class.
3. How do you separate those things that are part of the propositional content from those things on which an ExtrinsicMode externally depends? And can there be overlap between the two?
4. Is there a specific gUFO object property you could use to refer to the propostional content triples or would you need to create a custom one (as an instance from gufo:RelationshipType)?

**Comment by jpalmeida** [comment]
Hi @ManuelSimoes, we did not include in gUFO something akin to "Proposition". But in the gUFO documentation, we use propositions as examples of abstract individuals. This is a particular philosophical stance on propositions (which some of us have adopted in UFO-C and in some applications). When you adopt this stance, then, propositions are not externally dependent on the objects they denote (if any!).  So, having adopted this stance, whether you model something as an instrinsic mode or an extrinsic mode will depend on other factors (such as the dependence on a judicial court, or trial parties). 

Concerning 4, we do not have a specific gUFO object property for this.
